File: src/Aicl.Galapago.Model/Types/Ingreso/ComprobanteIngreso.cs
using System;
using System.ComponentModel.DataAnnotations;
using ServiceStack.DataAnnotations;
using ServiceStack.DesignPatterns.Model;

namespace Aicl.Galapago.Model.Types
{
	[Alias("COMPROBANTE_INGRESO")]
    [JoinTo(typeof(Sucursal),"IdSucursal", "Id", Order=0)]
    [JoinTo(typeof(Tercero),"IdTercero","Id", Order=1)]
    [JoinTo(typeof(Tercero),typeof(TipoDocumento),"IdTipoDocumento","Id", Order=2)]
    [JoinTo(typeof(PresupuestoItem),"IdCuentaReceptora","Id", Order=3)]
	public partial class ComprobanteIngreso:IHasId<Int32>,IHasPeriodo,IHasIdSucursal,IHasIdTercero
    {
		public ComprobanteIngreso(){}

		[Alias("ID")]
		[Sequence("COMPROBANTE_INGRESO_ID_GEN")]
		[PrimaryKey]
		[AutoIncrement]
		public Int32 Id { get; set;} 

		[Alias("FECHA")]
		public DateTime Fecha { get; set;} 

        [Alias("PERIODO")]
        [Required]
        [StringLength(6)]
        public String Periodo { get; set;} 

		[Alias("NUMERO")]
		public Int32 Numero { get; set;} 

		[Alias("DESCRIPCION")]
		[Required]
		[StringLength(50)]
		public String Descripcion { get; set;} 

		[Alias("VALOR")]
		[DecimalLength(15,2)]
		public Decimal Valor { get; set;} 

		[Alias("ID_CUENTA_RECEPTORA")]
		public Int32 IdCuentaReceptora { get; set;} 

		[Alias("ID_TERCERO")]
		public Int32 IdTercero { get; set;} 

		[Alias("ID_SUCURSAL")]
		public Int32 IdSucursal { get; set;} 

		[Alias("FECHA_ASENTADO")]
		public DateTime? FechaAsentado { get; set;} 

		[Alias("FECHA_ANULADO")]
		public DateTime? FechaAnulado { get; set;} 

		[Alias("EXTERNO")]
		public Boolean? Externo { get; set;} 

        [Alias("ID_TERCERO_RECEPTORA")]
        public Int32? IdTerceroReceptora { get; set;} 


        #region Sucursal
        [BelongsTo(typeof(Sucursal),"Nombre")]
        public string NombreSucursal{ get;set;}
        #endregion Sucursal

        #region Tercero
        [BelongsTo(typeof(Tercero),"Documento")]
        public string DocumentoTercero {get;set;}

        [BelongsTo(typeof(Tercero),"DigitoVerificacion")]
        public string DVTercero {get;set;}

        [BelongsTo(typeof(Tercero),"Nombre")]
        public string NombreTercero {get;set;}

        #endregion Tercero

        #region TipoDocumento Tercero
        [BelongsTo(typeof(TipoDocumento),"Nombre")]
        public string NombreDocumentoTercero {get;set;}
        #endregion TipoDocumento Tercero


        #region CuentaReceptora
        [BelongsTo(typeof(PresupuestoItem),"Codigo")]
        public String CodigoItem { get; set;} 

        [BelongsTo(typeof(PresupuestoItem),"Nombre")]
        public String NombreItem { get; set;} 
        #endregion CuentaReceptora

	}
}

File: src/Aicl.Galapago.Host.Web/WebApp/modules/app/model/ComprobanteIngreso.js
Ext.define('App.model.ComprobanteIngreso',{
	extend: 'Ext.data.Model',
	idProperty: 'Id',
	fields:[{
			name: 'Id',
			type: 'int'
		},{
			name: 'Fecha',
			type: 'date',
			convert: function(v){return Aicl.Util.convertToDate(v);}
		},{
			name: 'Periodo',
			type: 'string'
		},{
			name: 'Numero',
			type: 'int'
		},{
			name: 'Descripcion',
			type: 'string'
		},{
			name: 'Valor',
			type: 'number'
		},{
			name: 'IdCuentaReceptora',
			type: 'int'
		},{
			name: 'IdTercero',
			type: 'int'
		},{
			name: 'IdSucursal',
			type: 'int'
		},{
			name: 'FechaAsentado',
			type: 'date',
			convert: function(v){return Aicl.Util.convertToDate(v);}
		},{
			name: 'FechaAnulado',
			type: 'date',
			convert: function(v){return Aicl.Util.convertToDate(v);}
		},{
			name: 'Externo',
			type: 'boolean'
		},{
			name: 'IdTerceroReceptora',
			type: 'int'
		},{
			name: 'NombreSucursal',
			type: 'string'
		},{
			name: 'DocumentoTercero',
			type: 'string'
		},{
			name: 'DVTercero',
			type: 'string'
		},{
			name: 'NombreTercero',
			type: 'string'
		},{
			name: 'NombreDocumentoTercero',
			type: 'string'
		},{
			name: 'CodigoItem',
			type: 'string'
		},{
			name: 'NombreItem',
			type: 'string'
		}]
});